compiler/optimizer/OSRDefAnalysis.cpp

```diff
@@ -1254,34 +1467,42 @@ void TR_OSRLiveRangeAnalysis::buildOSRSlotSharingInfo(TR::Node *node, TR_BitVect
          }
       }
 
-   comp()->getOSRCompilationData()->ensureSlotSharingInfoAt(osrPoint->getByteCodeInfo());
+   comp()->getOSRCompilationData()->ensureSlotSharingInfoAt(bcInfo);
    }
 
 void TR_OSRLiveRangeAnalysis::buildOSRLiveRangeInfo(TR::Node *node, TR_BitVector *liveVars, TR_OSRPoint *osrPoint,
-   int32_t *liveLocalIndexToSymRefNumberMap, int32_t maxSymRefNumber, int32_t numBits,
-   TR_OSRMethodData *osrMethodData)
+   int32_t *liveLocalIndexToSymRefNumberMap, int32_t numBits, TR_OSRMethodData *osrMethodData)
    {
-   TR_ASSERT(liveVars, "live variable info must be available for a block\n");
+   int32_t byteCodeIndex = osrPoint->getByteCodeInfo().getByteCodeIndex();
+   bool allocated = false;
+
    _deadVars->setAll(numBits);
    *_deadVars -= *liveVars;
 
-   TR_BitVector *deadSymRefs;
-   if (_deadVars->isEmpty())
-      deadSymRefs = NULL;
-   else
+   TR_BitVector *deadSymRefs = NULL;
+   if (!_deadVars->isEmpty())
       {
-      deadSymRefs = new (trHeapMemory()) TR_BitVector(maxSymRefNumber+1, trMemory(), heapAlloc);
+      // Load the existing BitVector
+      deadSymRefs = osrMethodData->getLiveRangeInfo(byteCodeIndex);
+      if (!deadSymRefs)
+         {
+         deadSymRefs = new (trHeapMemory()) TR_BitVector(0, trMemory(), heapAlloc);
+         allocated = true;
```

Comment:
why do we need allocated rather than just testing the NULLness of the field?

Reply:
Or something similar

Reply:
Based on the prior implementation of this method, it was believed if there was at least one dead var, an allocation should be done. It has now been delayed until it is also confirmed that the var is one of interest.